ARCs/arc-0014.md

```diff
@@ -0,0 +1,67 @@
+---
+ams: 14
+title: Algorand standard for Authentication
+status: Draft
+---
+
+# Algorand Standard for authentication
+
+## Summary
+
+This document introduces the standard for authentication using the algorand accounts.
+
+## Abstract
+
+The goal of this standard is to define authorization process and authentication procedures for communication between web application and backend services or between two backend services. 
+
+## Specification
+
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**", and "**OPTIONAL**" in this document are to be interpreted as described in [RFC-2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+Base64 is the standard described in [RFC-4648](https://www.ietf.org/rfc/rfc4648.txt).
+
+## Request for comments
+
+Please comment or suggest pull request for this document here: [https://github.com/scholtz/AMS/issues](https://github.com/scholtz/AMS/issues)
+
+### Authentication
+
+Authentication is process in which person generates the authorization token for communication with backend service.
+
+Token MUST be generated from Algorand public and private key - from algorand address.
+
+In the first step of the authentication, the message to be signed is generated. The message MUST contain note field with configurable Realm. Realm is identifier of the service. It MAY be the web address of the service for intended purpose.
+
+The message MUST be self signed - the receiver is the same as the sender.
+
+The message MUST have network parameters filled in. The network genesis hash SHOULD be configured for validation pursposes by backend service. The latest block is the time of expiration of the token. The latest block MUST be provided.
+
+In the second step, the message MUST be signed by the account private key.
+
+The message MUST not be submitted to the network (mainnet, testnet) nor should be published in any other way. It may be used only for communcation to the backend service.
+
+### Authorization
+
+Authorization determines what resources a user can access.
+
+Authorization is the process of validating authorization token by the backend application and allowing the account permissions to the backend services.
+
+Authorization token SHOULD be send to backend service using the authorization header.
```

Comment:
Think we should not define it as a "backend service" but any verifier. The verifier could be a backend service or some front end protected service

Reply:
yes, I agree the param for this should be verifier as it allows the definition to vary from backend or front end sec